field_operations/update-all-fields.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * Sets all non-title fields to optional.
+ */
+function hook_update_N() {
+  // Services we need.
+  $entityFieldManager = \Drupal::service('entity_field.manager');
+  $config = \Drupal::service('config.factory');
+
+  // Get all node fields.
+  $fields = $entityFieldManager->getFieldMap();
+  $fields = $fields['node'];
+
+  // Pop off the entity key & generic fields (nid, uuid, revision_log, etc).
+  $fields = array_slice($fields, 18, null, true);
+
```

Comment:
18 is a magic number, because when I var_dump($fields), the first 18 results for node fields are:
```
[0]=>
  string(3) "nid"
  [1]=>
  string(4) "uuid"
  [2]=>
  string(3) "vid"
  [3]=>
  string(8) "langcode"
  [4]=>
  string(4) "type"
  [5]=>
  string(5) "title"
  [6]=>
  string(3) "uid"
  [7]=>
  string(6) "status"
  [8]=>
  string(7) "created"
  [9]=>
  string(7) "changed"
  [10]=>
  string(7) "promote"
  [11]=>
  string(6) "sticky"
  [12]=>
  string(18) "revision_timestamp"
  [13]=>
  string(12) "revision_uid"
  [14]=>
  string(12) "revision_log"
  [15]=>
  string(29) "revision_translation_affected"
  [16]=>
  string(16) "default_langcode"
  [17]=>
  string(4) "path"
```

I think there is a isBaseField() method somewhere that would probably be better to check against, but I was having trouble calling it in this case.